tasks/check_primary_cert.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+declare PT__installdir
+# shellcheck disable=SC1090
+source "$PT__installdir/ca_extend/files/common.sh"
+PUPPET_BIN='/opt/puppetlabs/puppet/bin'
+
+expiry_date="$($PUPPET_BIN/openssl x509 -enddate -noout -in "$($PUPPET_BIN/puppet config print hostcert)")" || {
```

Comment:
This could be a failure point. If they have already tried to run the plan with a regeneration of the primary's cert, it would have been deleted. There is no way to rerun the plan in this case. We should allow for moving forward if this check fails.

Reply:
This task is not run if `$regen_primary_cert` is `true`.

Reply:
If someone runs it with `regen_primary_cert = false` and the cert has already been cleaned it will fail with no clear method of moving forward. The failure could point to running it with `regen_primary_cert = true` to get around the failure.

Reply:
Ah, that's not a scenario I thought of.  I'll add a check for the file and a useful error message.

Reply:
Also added in `2a3899a`